examples/transmission_main_example.py

```diff
@@ -77,10 +61,63 @@ def main(args):
     axis('off')
     show()
 
+
+def main(args):
+    input_filename = str(args.filename)
+    split_filename = input_filename.split(".")
+    json_filename = split_filename[0]+'.json'
+    try:
+        assert split_filename[1] in ('json','xls','csv','xlsm')
+    except AssertionError as e:
+        print(f'invalid file extension .{split_filename[1]}')
+        raise e
+    if split_filename[1] != 'json':
+        print(f'parse excel input to {json_filename}')
+        convert_file(input_filename)
+
+    json_data = load_json(json_filename)
+    read_eqpt_library(EQPT_LIBRARY_FILENAME)
+
+    network = network_from_json(json_data)
+    build_network(network)
+
+    spacing = 0.05 #THz
+    si = SpectralInformation() # !! SI units W, Hz
+    si = si.update(carriers=tuple(Channel(f, (191.3+spacing*f)*1e12, 
+            32e9, 0.15, Power(1e-3, 0, 0)) for f in range(1,97)))
+
+    trx = [n for n in network.nodes() if isinstance(n, Transceiver)]
+    if args.list>=1:
```

Comment:
(Copied from previous PR review):

I don't think listing all the network nodes should be exclusive with running a simulation.

If that's the case, there should be a greater separation between these two modalties.

Also, if this is the case, return early rather than indenting everything below.

Reply:
Sorry Jame, I don't find this comment helpful, 
especially as it is the 2nd time I'm pulling this part of the code.
You certainly don't want to list network nodes AND run the simulation. You are just looking for the nodes that will be the source and destination or your simulation.

Reply:
Okay, we can keep these flags exclusive with each other.

We should push this modality out of this particular function into the `if __name__ == "__main__"` block.

Reply:
ok,
for future improvement